sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/IndexActionConverter.java

```diff
@@ -66,19 +57,16 @@
             indexAction.setActionType(actionType);
         }
 
-        Map<String, Object> additionalProperties;
-        TypeReference<Map<String, Object>> typeRef = new TypeReference<Map<String, Object>>() {};
 
         Map<String, Object> mapProperties = PrivateFieldAccessHelper.get(obj, "properties", Map.class);
-        if (mapProperties != null) {
-            DYNAMIC_TYPE_MAPPER.setSerializationInclusion(JsonInclude.Include.ALWAYS);
-            additionalProperties = DYNAMIC_TYPE_MAPPER.convertValue(mapProperties, typeRef);
-        } else {
+        if (mapProperties == null) {
             T properties = obj.getDocument();
-            additionalProperties = STRONGLY_TYPE_MAPPER.convertValue(properties, typeRef);
+            ByteArrayOutputStream sourceStream = serializer.serialize(new ByteArrayOutputStream(), properties);
+            mapProperties = serializer.deserialize(new ByteArrayInputStream(sourceStream.toByteArray()),
+                new TypeReference<Map<String, Object>>() { });
```

Comment:
Let's make this a static somewhere instead of instantiating a new instance every time we are deserializing. I believe `TypeReference<Map<String, Object>>` is used in a few places so let's make it a constant somewhere in implementation.

Reply:
Will place it into impl Utility.